thetis/callback.py

```diff
@@ -481,3 +481,49 @@ def __call__(self):
             field_vals.append(np.reshape(self._evaluate_field(field_name), (ndetectors, -1)))
 
         return np.hstack(field_vals)
+
+
+class FunctionalCallback(DiagnosticCallback):
+    """Base class for callbacks that evaluate a functional involving integrals in both time and space. This callback
+    can also be used to assemble time dependent objective functionals for adjoint simulations. Time integration is
+    achieved using the trapezium rule."""
+    variable_names = ['current integral', 'objective value']
+
+    def __init__(self, scalar_callback, solver_obj, **kwargs):
+        """
+        Creates error comparison check callback object
```

Comment:
What?

Reply:
Sorry, this is something I forgot to change from a previous version